test/jdk/jdk/nio/zipfs/TestLocOffsetFromZip64EF.java

```diff
@@ -143,88 +134,87 @@ public void walkZipFileTest() throws IOException {
     }
 
     /**
-     * Create the files that will be added to the ZIP file
-     * @throws IOException if there is a problem  creating the files
+     * This produces a ZIP with similar features as the one created by 'Info-ZIP' which
+     * caused 'Extended timestamp' parsing to fail before JDK-8255380.
+     *
+     * The issue was sensitive to the ordering of 'Info-ZIP extended timestamp' fields and
+     * 'Zip64 extended information' fields. ZipOutputStream and 'Info-ZIP' order these differently.
+     *
+     * ZipFileSystem tried to read the Local file header while parsing the extended timestamp,
+     * but if the Zip64 extra field was not read yet, ZipFileSystem would incorrecly try to read
+     * the Local File header from offset 0xFFFFFFFF.
+     *
+     * This method creates a ZIP file which includes a CEN with the following features:
+     *
+     * - Its extra field has a 'Info-ZIP extended timestamp' field followed by a
+     *   'Zip64 extended information' field.
+     * - The sizes and offset fields values of the CEN are set to 0xFFFFFFFF (Zip64 magic values)
+     *
      */
-    private static void createFiles() throws IOException {
-        try (RandomAccessFile file = new RandomAccessFile(LARGE_FILE_NAME, "rw")
-        ) {
-            System.out.printf("Creating %s%n", LARGE_FILE_NAME);
-            file.setLength(LARGE_FILE_SIZE);
-            System.out.printf("Creating %s%n", SMALL_FILE_NAME);
-            Files.writeString(Path.of(SMALL_FILE_NAME), "Hello");
+    public void createZipWithZip64Ext() throws IOException {
+        ByteArrayOutputStream out = new ByteArrayOutputStream();
+        try (ZipOutputStream zo = new ZipOutputStream(out)) {
+
+            ZipEntry e = new ZipEntry("entry");
+            // Make it STORED and empty to simplify parsing
+            e.setMethod(ZipEntry.STORED);
+            e.setSize(0);
+            e.setCrc(0);
+
+            // Make ZipOutputStream output an 'Info-Zip extended timestamp' extra field
+            e.setLastModifiedTime(FileTime.from(Instant.now()));
+            e.setLastAccessTime(FileTime.from(Instant.now()));
+
+            // Add an opaque extra field, right-sized for a Zip64 extended field
+            // We'll update this below
+            byte[] zip64 = new byte[ZIP64_SIZE];
+            ByteBuffer buffer = ByteBuffer.wrap(zip64).order(ByteOrder.LITTLE_ENDIAN);
+            buffer.putShort(UNKNOWN_TAG); // Opaque tag makes ZipEntry.setExtra ignore it
+            buffer.putShort((short) (zip64.length - 2 * Short.BYTES)); // Data size
```

Comment:
The length of the zip64 extra field is going to be 28.   I would suggest making this a bit clearer and adding a field/constant which is set to 28 for clarity to make it easier for those who might look at this code later

Reply:
> The length of the zip64 extra field is going to be 28. I would suggest making this a bit clearer and adding a field/constant which is set to 28 for clarity to make it easier for those who might look at this code later

I believe this should be well-described now between the constants and the `makeOpaqueExtraField` method.